.github/workflows/wheels-build.yml

```diff
@@ -0,0 +1,170 @@
+          - { ctk: '12.0.1', arch: 'arm64', python: '3.10', linux_ver: 'ubuntu20.04' }
+          "
+
+          echo "MATRIX=$(
+            yq -n -o json 'env(MATRIX)' | \
+            jq -c '${{ inputs.matrix_filter }} | {include: .}' \
+          )" | tee --append "${GITHUB_OUTPUT}"
+
+  wheel-build:
+    name: cibuildwheel ${{ matrix.arch }} ${{ matrix.python }} ${{ matrix.ctk }}
+    needs: [setup-env-vars, compute-matrix]
+    strategy:
+      matrix: ${{ fromJSON(needs.compute-matrix.outputs.MATRIX) }}
+    runs-on: "linux-${{ matrix.arch }}-${{ inputs.node_type }}"
+    container:
+      image: "rapidsai/ci-wheel:cuda${{ matrix.ctk }}-${{ matrix.linux_ver }}-py${{ matrix.python }}"
+
+      env:
+        RAPIDS_BUILD_TYPE: ${{ inputs.build_type }}
+        RAPIDS_PY_VERSION: ${{ matrix.python }}
+        RAPIDS_EPOCH_TIMESTAMP: ${{ needs.setup-env-vars.outputs.RAPIDS_EPOCH_TIMESTAMP }}
+        PIP_EXTRA_INDEX_URL: ${{ needs.setup-env-vars.outputs.PIP_EXTRA_INDEX_URL }}
+
+    steps:
+      - uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ vars.AWS_ROLE_ARN }}
+          aws-region: ${{ vars.AWS_REGION }}
+          role-duration-seconds: 43200 # 12h
+      - name: checkout code repo
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ inputs.repo }}
+          ref: ${{ inputs.sha }}
+          fetch-depth: 0 # unshallow fetch for setuptools-scm
+          persist-credentials: false
+
+      - name: Standardize repository information
+        run: |
+          echo "RAPIDS_REPOSITORY=${{ inputs.repo || github.repository }}" >> "${GITHUB_ENV}"
+          echo "RAPIDS_SHA=$(git rev-parse HEAD)" >> "${GITHUB_ENV}"
+          echo "RAPIDS_REF_NAME=${{ inputs.branch || github.ref_name }}" >> "${GITHUB_ENV}"
+          echo "RAPIDS_NIGHTLY_DATE=${{ inputs.date }}" >> "${GITHUB_ENV}"
+
+      - name: Preprocess extra repos
+        id: preprocess-extras
+        if: ${{ inputs.extra-repo != '' }}
+        run: |
+          EXTRA_REPO_PATH=$(echo ${{ inputs.extra-repo }} | cut -d "/"  -f 2)
+          echo "EXTRA_REPO_PATH=${EXTRA_REPO_PATH}" >> $GITHUB_OUTPUT
+
+      - name: checkout extra repos
+        uses: actions/checkout@v3
+        if: ${{ inputs.extra-repo != '' }}
+        with:
+          repository: ${{ inputs.extra-repo }}
+          ref: ${{ inputs.extra-repo-sha }}
+          path: "./${{ steps.preprocess-extras.outputs.EXTRA_REPO_PATH }}"
+          ssh-key: ${{ secrets[inputs.extra-repo-deploy-key] }}
+          persist-credentials: false
+
+      - name: Build and repair the wheel
+        run: |
+          # Store internal pypi credentials before any step that may download wheels
+          printf 'machine pypi.k8s.rapids.ai\n\tlogin cibuildwheel\n\tpassword ${{ secrets.RAPIDSAI_PYPI_CI_PASSWORD }}\n' > ~/.netrc
+          ${{ inputs.build_script }}
+        env:
+          GH_TOKEN: ${{ github.token }}
```

Comment:
What is the token needed for? The private repo clone still happens in this workflow, is there another purpose?

Reply:
Not sure, `conda-cpp-build.yaml` also uses it so I coped it

Reply:
Hmm looks like it was added in https://github.com/rapidsai/shared-action-workflows/pull/87 due to https://github.com/rapidsai/gha-tools/pull/53. Perhaps @ajschmidt8 could provide extra info, but I'm not sure that's needed for us because we never run `rapids-env-update`.

Reply:
I think it's fine to leave this in. The `gh` CLI tool can't be used without it. At some point someone may want to use the `gh` CLI tool in one of their scripts, at which point we'd have to add it back.

> Hmm looks like it was added in #87 due to rapidsai/gha-tools#53. Perhaps @ajschmidt8 could provide extra info, but I'm not sure that's needed for us because we never run rapids-env-update.

This is actually another opportunity for deduplication.

The `RAPIDS_DATE_STRING` variable and the `wheel-epoch-timestamp` job in this workflow are effectively doing the same thing: generating a unique per-workflow timestamp that can be used throughout the jobs in a given workflow.

I'm partial to using the `RAPIDS_DATE_STRING` method, which uses the `gh` CLI tool to get the timestamp of the start time of the currently running workflow. This method eliminates the need for an extra GHA job, which simplifies the workflow itself and the corresponding workflow graph that's generated in the GHAs web GUI.

I don't think we necessarily need to dedupe these implementations in this PR (though feel free if you think it's doable), but at the very least we should probably make a note of it.

@divyegala, @vyasr, what are your thoughts?

Reply:
@ajschmidt8 and I discussed this offline, and we think we can deduplicate and just create a small tool for `RAPIDS_DATE_STRING`. @vyasr this will also solve the problems we were discussing yesterday with producing `RAPIDS_EPOCH_TIMESTAMP` for local builds